packages/x-date-pickers/src/internals/components/PickersViewLayout/PickersViewLayout.tsx

```diff
@@ -99,18 +127,49 @@ export const PickersViewLayout = React.forwardRef(function PickersViewLayout<
             disabled={disabled}
             readOnly={readOnly}
           />
-        )}
+        </PickersViewLayoutToolbar>
+      ) : null,
+    content: view && (
+      <PickersViewLayoutContent className={classes.content} key="MUI-X_content">
         {!!Tabs && <Tabs view={view} onViewChange={onViewChange} {...componentsProps?.tabs} />}
         {children}
       </PickersViewLayoutContent>
-      <ActionBar
-        onAccept={onAccept}
-        onClear={onClear}
-        onCancel={onCancel}
-        onSetToday={onSetToday}
-        actions={wrapperVariant === 'desktop' ? [] : ['cancel', 'accept']}
-        {...componentsProps?.actionBar}
-      />
-    </PickersViewLayoutRoot>
-  );
+    ),
+    actionBar: (
+      <PickersViewLayoutActionBar className={classes.actionbar} key="MUI-X_actionBar">
+        <ActionBar
+          onAccept={onAccept}
+          onClear={onClear}
+          onCancel={onCancel}
+          onSetToday={onSetToday}
+          actions={wrapperVariant === 'desktop' ? [] : ['cancel', 'accept']}
+          {...componentsProps?.actionBar}
+        />
+      </PickersViewLayoutActionBar>
+    ),
+  };
+
+  const Layout = components?.Layout ?? DefaultPickersViewLayout;
+  const layoutProps = useSlotProps<
+    React.JSXElementConstructor<PickersViewLayoutSlotOwnerState<any, any>>,
+    PickersViewLayoutProps<any, any>,
+    any,
+    any
+  >({
+    elementType: Layout,
+    externalSlotProps: componentsProps?.layout,
+    additionalProps: {
+      ref,
+      className: clsx(className, classes.root),
+      ...subComponents,
+    },
+    externalForwardedProps: other,
+    ownerState: { ...props, ...subComponents },
```

Comment:
I don't have the strongest opinion on this approach, but I tend to lean towards maybe having a render function instead of this yet another approach to customize rendering. 🙈 🤷 
Or am I mistaken and we do already have similar approach in other place(s)? 🤔

Reply:
We already use the `ownerState` of the slot to pass information (the `previousIconButton` slot for instance has a `hidden` property added on top of the props).

But I never saw React nodes passed their.
My idea was to re-use an already existing customization feature (the callback version of the slot props).
But maybe that's too hacky.
Maybe it would be interesting writing an example with both approaches to pick the one we prefer.

Reply:
I'm not sure about what would be a "render function" interface.

```jsx
const renderLayout = (props, Toolbar, Content, ActionBar, Shortcuts) => <>
    <Toolbar />
    <Content />
    <ActionBar />
    <Shortcuts />
  </>

<DatePicker
  renderLayoutContent={renderLayout}
  // Or
  componentProps={{ layout: { renderLayout } }}
/>
```

It does not seems to be less hacky


----

A last option I see could be to do not fully respect usual slot behavior. Instead of replacing the `PickersViewLayout` when `components.Layout` is provided, we replace it's content.

```jsx
const PickersViewLayout = (props) => {
	const { components, componentsProps, ...other} = props;

	// We still create the default components based on the props
	const ActionBar = components?.ActionBar ?? PickersActionBar;
	const actionBarProps = {onAccept: other.onAccept, onClear: other.onClear, ..., ...componentsProps?.actionBar}

	// ...
	
	if(components?.Layout) {
		// If the layout is overridden, we provide components and components props to writ every thing simply
		const Layout = components.Layout;
		return <Layout {...props} components={{ActionBar, Toolbar, ...}} componentsProps={{ actionBar: actionBarProps, ...}}/>
	}

	// Otherwise we do as usual
	return <LayoutRoot >
		{/* Defautl content */ }
	</LayoutRoot>

}
```


From a DX point of view I tend to prefer the initial solution with children. It's hacky but we are at the limit where a headless approach could make sense. So hacky solution for dev who don't have time and headless for clean solution sounds like a good trade off

Reply:
I agree that your solution is less hacky